infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -34,7 +37,8 @@ public AbstractServer() {
      */
     public void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         if (isShutdown()) return;
-        if (!getHandler().handle(msg, ctx, r)) {
+        boolean isMetricsEnabled = MetricsUtils.isMetricsCollectionEnabled();
+        if (!getHandler().handle(msg, ctx, r, isMetricsEnabled)) {
```

Comment:
See above (ops "hair on fire") argument for the consistency of related stats. Many times, metrics values will be evaluated by calm people, and the extra arg to make all-or-nothing metrics keeping is wasted effort.  I'd rather add the argument to avoid creating distractions for stressed ops people who don't need more stress.